app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params) # (username: params[:username], email: params[:email], password: params[:password])
+
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.save
+      @user.update(user_params)
```

Comment:
You don't need to save before making before making the update, use update_attributes method

Reply:
changes made....